src/test/java/com/mongodb/hibernate/jdbc/SessionFactoryTests.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.jdbc;
+
+import static org.hibernate.cfg.JdbcSettings.JAKARTA_JDBC_URL;
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.mongodb.hibernate.exception.ConfigurationException;
+import java.util.Map;
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.service.spi.ServiceException;
+import org.junit.jupiter.api.Test;
+
+class SessionFactoryTests {
+
+    @Test
+    void test_success() {
+        buildSessionFactory(Map.of(JAKARTA_JDBC_URL, "mongodb://localhost/test"));
+    }
+
+    @Test
+    void test_invalid_connection_String() {
+        var exception = assertThrows(
+                ServiceException.class,
+                () -> buildSessionFactory(Map.of(JAKARTA_JDBC_URL, "jdbc:postgresql://localhost/test")));
+        assertInstanceOf(ConfigurationException.class, exception.getCause());
+    }
+
+    @Test
+    void test_when_database_absent() {
+        var exception = assertThrows(
+                ServiceException.class, () -> buildSessionFactory(Map.of(JAKARTA_JDBC_URL, "mongodb://localhost")));
+        assertInstanceOf(ConfigurationException.class, exception.getCause());
+    }
+
+    private void buildSessionFactory(Map<String, String> configurationValues) throws ServiceException {
+        var cfg = new Configuration();
+        configurationValues.forEach(cfg::setProperty);
+        try (SessionFactory ignored = cfg.buildSessionFactory()) {
+            // no-op
+        }
+    }
+}
```

Comment:
the testing only covers some basic branches of jdbc URL, not sure how to test user and password or whether it is needed.

Reply:
It's needed but to test it we will need more configuration in CI to start a cluster with authentication enabled.